packages/primitives/src/Switch/Switch.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react'
+import styled from '@emotion/styled'
+import { Theme, css } from 'theme-ui'
+import { SerializedStyles } from '@emotion/serialize'
+import {
+  StyledSystemProps,
+  variant,
+  sx,
+  shouldForwardProp,
+  styledSystemProps,
+  Box,
+} from '../Box/Box'
+import { Label } from '../Label'
+import { Input } from '../Input'
+
+function SwitchIndicator({
+  checked,
+  disabled,
+}: {
+  checked?: boolean
+  disabled?: boolean
+}) {
+  return (
+    <Box
+      as="span"
+      border="solid"
+      borderWidth={1}
+      opacity={disabled ? 0.7 : 1}
+      backgroundColor={checked ? 'blue.1' : 'grey.5'}
+      borderColor={checked ? 'blue.4' : 'grey.5'} // darkest blue
+      borderRadius={40}
+      display="inline-block"
+      height={16}
+      ml={-38}
+      mr={2}
+      position="relative"
+      sx={{
+        transition: 'background-color 0.1s cubic-bezier(0.4, 1, 0.75, 0.9)',
+        ...(disabled && { cursor: 'not-allowed', outline: 'none' }),
+
+        'input:focus ~ &': {
+          outline: '2px auto',
+          outlineColor: 'blue.2',
+          outlineOffset: 2,
+        },
+
+        '&::before': {
+          background: 'white',
```

Comment:
This should be pulled from the theme variant as well in case someone wants to change the bg color for this.

Reply:
Sounds good, I'll setup the variant to setup the background and other colors 👍